### PR TITLE
Fix some panics + add testing and fix for map sharing 

### PR DESCRIPTION
**Pull request by astoycos**
See commit messages for details.

### CODE REVIEW

**Comment by astoycos** [comment]
CI will fail here until I manually push images OR we merge broken

File: bpfd/src/oci_utils/image_manager.rs [hunk 1]
```diff
@@ -493,7 +494,9 @@ fn load_image_meta(image_content_path: PathBuf) -> Result<ContainerImageMetadata
     let file_content =
         read_to_string(image_config_path).context("failed to read image config file")?;
 
-    let image_config: Value = serde_json::from_str(&file_content).unwrap();
+    // This should panic since it means that the on disk memory format has changed during runtime.
```
Comment (dave-tucker) [inline bpfd/src/oci_utils/image_manager.rs:497]: on disk memory? you mean on disk storage format?

**Review by dave-tucker — CHANGES_REQUESTED** [review]
One nit, and were you also planning to add some documentation around the LIBBPF_PIN_BY_NAME flag?

File: bpfd/src/oci_utils/image_manager.rs (hunk 1 above)
Comment (astoycos) [reply bpfd/src/oci_utils/image_manager.rs]: Lol yerp thanks!

**Comment by astoycos** [comment]
> One nit, and were you also planning to add some documentation around the LIBBPF_PIN_BY_NAME flag?

Ah yeah I'll do this, we definitely need an interim note in the docs

**Review by dave-tucker — APPROVED** [review]

**Pull request merged**